src/naive/date.rs

```diff
@@ -77,10 +77,10 @@ impl NaiveWeek {
     #[inline]
     #[must_use]
     pub fn first_day(&self) -> NaiveDate {
-        let start = self.start.num_days_from_monday();
-        let end = self.date.weekday().num_days_from_monday();
-        let days = if start > end { 7 - start + end } else { end - start };
-        self.date - Duration::days(days.into())
+        let start = self.start.num_days_from_monday() as i32;
+        let ref_day = self.date.weekday().num_days_from_monday() as i32;
+        let days = if start > ref_day { start - ref_day - 7 } else { start - ref_day };
```

Comment:
Please add a comment here and in `last_day()` about the overflow/underflow risks.

Also, please extract the common `start - ref_day` expression for both methods.

Reply:
> Also, please extract the common `start - ref_day` expression for both methods.

I don't think extracting it to a common methods it going to make it more readable. To compare:
```rust
let days = if start > ref_day { start - ref_day - 7 } else { start - ref_day };
```
And `last_day()`:
```rust
let days = if end < ref_day { end - ref_day + 7 } else { end - ref_day };
```

Reply:
Added a comment.

Reply:
Sorry, I just meant the expression:

```rust
let days = end - ref_day - if end < ref_day { 7 } else { 0 };
```

Reply:
Ah! That is indeed better.